libraries/freertos_plus/standard/tls/src/iot_tls.c

```diff
@@ -852,41 +852,41 @@ BaseType_t TLS_Recv( void * pvContext,
 
     if( ( NULL != pxCtx ) && ( pdTRUE == pxCtx->xTLSHandshakeSuccessful ) )
     {
-        while( xRead < xReadLength )
+        /* The secure sockets receive API does not guarantee that the bytes requested is the bytes
```

Comment:
It might be more clear to just say what's in your last sentence:

"This routine immediately returns the bytes received from mbedtls_ssl_read (if any) unless MBEDTLS_ERR_SSL_WANT_READ is returned, in which case we try again."